src/NServiceBus.Core/Persistence/Msmq/SubscriptionStorage/MsmqSubscriptionStorageQueue.cs

```diff
@@ -51,6 +52,7 @@ public void TryReceiveById(string messageId)
         {
             try
             {
+                //todo: add failing test before fixing this
```

Comment:
@timbussmann turns out that MessageQueueTransactionType.None works for both types of queues (see tests added). Should we leave as is with a comment?